app/flatpak-builtins-build.c

```diff
@@ -434,27 +410,27 @@ flatpak_builtin_build (int argc, char **argv, GCancellable *cancellable, GError
         }
 
       *split++ = 0;
-      add_args (argv_array,
-                "--bind", split, opt_bind_mounts[i],
-                NULL);
+      flatpak_bwrap_add_args (bwrap,
+                              "--bind", split, opt_bind_mounts[i],
+                              NULL);
     }
 
   if (opt_build_dir != NULL)
     {
-      add_args (argv_array,
-                "--chdir", opt_build_dir,
-                NULL);
+      flatpak_bwrap_add_args (bwrap,
+                              "--chdir", opt_build_dir,
+                              NULL);
     }
 
-  g_ptr_array_add (argv_array, g_strdup (command));
+  g_ptr_array_add (bwrap->argv, g_strdup (command));
```

Comment:
(Style) Up until here all access to bwrap internals is neatly encapsulated, but here the code is more similar to the original.

What's is/was the essence of this code?  Might it be worth a separate function?

In any case I imagine these g_ptr_array_add() could be calls to flatpak_bwrap_add_args() instead for consistency?  That will have the vararg overhead, but it seems more consistent.

Reply:
I tried to only change things that were obviously right, as this is is more like a first step, but you're right, in this case we could just use flatpak_bwrap_add_args()